### ISSUE TITLE
Magic Code Required for webchat V4.11.0

### ISSUE THREAD

**Issue by CharingJC** [microsoft/BotFramework-WebChat#3794]
## Version

4.11.0

## Describe the bug
0. Team have taken following steps. In order not to have 6 digits when User Sign in AD to talk to BOT and can use it.
we follow this link https://blog.botframework.com/2018/09/01/using-webchat-with-azure-bot-services-authentication/ to get rid of the 6 digital verification code (aka magic code). 

But on 08 February 2021, some users have access to BOT with Microsoft Edge Version 88.0.705.63 and when logged in to talk to Bot and found that it has 6 digits

1.	In my test environment, I was asked to copy and paste the magic code. And I noticed the error message with ‘getSessionId’ component.

  
![image](https://user-images.githubusercontent.com/74894646/111113882-6c674900-859d-11eb-8705-992037128b0d.png)

2.	Here is the source code about getting session id from the webchat.js
[source code](https://github.com/microsoft/BotFramework-WebChat/blob/a972426d7f69bf74d2f474c3943c54dc50b09a85/packages/api/src/hooks/Composer.js
)
 
![image](https://user-images.githubusercontent.com/74894646/111113888-6ec9a300-859d-11eb-88b6-8e032efecfde.png)

## Steps to reproduce

I am using the following code to connect to the bot service on azure.
```
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>Web Chat: Full-featured bundle</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!--
      This CDN points to the latest official release of Web Chat. If you need to test against Web Chat's latest bits, please refer to using Web Chat's latest bits:
      https://github.com/microsoft/BotFramework-WebChat#how-to-test-with-web-chats-latest-bits
    -->
    <script crossorigin="anonymous" src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script>
    <style>
      html,
      body {
        height: 100%;
      }

      body {
        margin: 0;
      }

      #webchat {
        height: 100%;
        width: 100%;
      }

      .webchat__upload-button {
        display: none !important;
      }
    </style>
  </head>
  <body>
    <div id="webchat" role="main"></div>
    <script>
      (
        async function () {
        // In this demo, we are using Direct Line token from MockBot.
        // Your client code must provide either a secret or a token to talk to your bot.
        // Tokens are more secure. To learn about the differences between secrets and tokens
        // and to understand the risks associated with using secrets, visit https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication?view=azure-bot-service-4.0
        const userId = 'dl_' + Math.random().toString(36).substr(2, 9); 
        var myHeaders = new Headers();
        myHeaders.append("Authorization", "Bearer " + YOUR_TOKEN);
        myHeaders.append("Content-Type", "application/json");

        var raw = JSON.stringify({"User":{"Id":userId}});

        var requestOptions = {
          method: 'POST',
          headers: myHeaders,
          body: raw,
          redirect: 'follow'
        };

        const response = await fetch("https://directline.botframework.com/v3/directline/tokens/generate", requestOptions)
        .then(function(serverPromise){ 
      serverPromise.json()
        .then(function(j) { 
          const token = j.token
        window.WebChat.renderWebChat(
          {
            styleOptions: {
               hideUploadButton: true
            },
            locale: 'zh-Hans',
            directLine: window.WebChat.createDirectLine({ token })
          },
          document.getElementById('webchat')
        );

        document.querySelector('#webchat > *').focus(); 
        })
        .catch(function(e){
          console.log(e);
        });
        })
        .catch(function(e){
            console.log(e);
          });
        
        
      })().catch(err => console.error(err));
    </script>
  </body>
</html>

```

## Expected behavior

sign in without using magic code.

## Additional context

_Because of the issues with the previous approach, Azure Bot Service removed the need for the magic code. Azure Bot Service guarantees that the sign-in process can only be completed in the same browser session as the Web Chat itself. To enable this protection, as a bot developer, you must start Web Chat with a Direct Line token that contains a list of trusted domains that can host the bot's Web Chat client. Before, you could only obtain this token by passing an undocumented optional parameter to the Direct Line token API. Now, with enhanced authentication options, you can statically specify the trusted domain (origin) list in the Direct Line configuration page._

_Some channels such as Microsoft Teams, Direct Line, and WebChat are able to do this without the user noticing. For example, WebChat uses session cookies to ensure that the sign-in flow took place in the same browser as the WebChat conversation. However, for other channels the user is often presented with a 6-digit magic code. This is similar to a built-in multi-factor authentication, as the Bot Framework Token Service will not release the token to the bot unless the user finishes the final authentication, proving that the person who signed-in has access to the chat experience by entering the 6-digit code._

More Information:
[doc1](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication?view=azure-bot-service-4.0)
[doc2](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-concept-authentication?view=azure-bot-service-4.0)
[doc3](https://blog.botframework.com/2018/09/01/using-webchat-with-azure-bot-services-authentication/)
[Bug]

**Comment by CharingJC** [comment]
Hello @daveta 

Hope this message find you well.
Any update on this issue?

BR,
Juncheng

**Comment by stevkan** [comment]
@CharingJC, A few questions:

- Was there an upgrade to Edge on or around that day?
- Also, does this issue come up in other browsers or is it isolated to Edge, alone?
- You also state "some users" are experiencing this. Is this intermittent in who and when someone is affected? Or, does it always happen but only to specific users?

There is a newer version of Edge available (version 89.0.774.54 (Official build)). Does it occur with this version, as well?

**Comment by CharingJC** [comment]
> @CharingJC, A few questions:
> 
> * Was there an upgrade to Edge on or around that day?
> * Also, does this issue come up in other browsers or is it isolated to Edge, alone?
> * You also state "some users" are experiencing this. Is this intermittent in who and when someone is affected? Or, does it always happen but only to specific users?
> 
> There is a newer version of Edge available (version 89.0.774.54 (Official build)). Does it occur with this version, as well?

Hello stevkan,

Thank you for the response. And here are the answers.

_Was there an upgrade to Edge on or around that day?_ 
Edge V88 was released on 21-Jan-2021


_Also, does this issue come up in other browsers or is it isolated to Edge, alone?_ 
Edge V89 and Chrome V89 also have the same issue

_You also state "some users" are experiencing this. Is this intermittent in who and when someone is affected? Or, does it always happen but only to specific users?_ There is no working scenario so far.

BR,
CharingJC

**Comment by CharingJC** [comment]
> > @CharingJC, A few questions:
> > 
> > * Was there an upgrade to Edge on or around that day?
> > * Also, does this issue come up in other browsers or is it isolated to Edge, alone?
> > * You also state "some users" are experiencing this. Is this intermittent in who and when someone is affected? Or, does it always happen but only to specific users?
> > 
> > There is a newer version of Edge available (version 89.0.774.54 (Official build)). Does it occur with this version, as well?
> 
> Hello stevkan,
> 
> Thank you for the response. And here are the answers.
> 
> _Was there an upgrade to Edge on or around that day?_
> Edge V88 was released on 21-Jan-2021
> 
> _Also, does this issue come up in other browsers or is it isolated to Edge, alone?_
> Edge V89 and Chrome V89 also have the same issue
> 
> _You also state "some users" are experiencing this. Is this intermittent in who and when someone is affected? Or, does it always happen but only to specific users?_
> There is no working scenario so far.
> 
> BR,
> CharingJC

Hi Steven,

Hope that your week is off to a great start.
I would appreciate your insights. Please send me a mail when you are available.

BR,
Juncheng

**Comment by CharingJC** [comment]
@stevkan Dear Steven,

Any update?

**Comment by stevkan** [comment]
@CharingJC, please verify that 'Enhanced authentication' is enabled in the Direct Line channel settings for the bot in Azure. If this is not enabled, which is a key part to bypassing needing a magic code, then the `getsessionid` error is produced. Refer to Step 5 of 'Enabling Directline' in [Using WebChat with Azure Bot Service’s Authentication](https://blog.botframework.com/2018/09/01/using-webchat-with-azure-bot-services-authentication/). In my quick testing, if I ensure 'Enhanced authentication' is enabled, the error disappears.

If the issue persists with 'Enhanced authentication' enabled, please let me know.

**Comment by CharingJC** [comment]
> @CharingJC, please verify that 'Enhanced authentication' is enabled in the Direct Line channel settings for the bot in Azure. If this is not enabled, which is a key part to bypassing needing a magic code, then the `getsessionid` error is produced. Refer to Step 5 of 'Enabling Directline' in [Using WebChat with Azure Bot Service’s Authentication](https://blog.botframework.com/2018/09/01/using-webchat-with-azure-bot-services-authentication/). In my quick testing, if I ensure 'Enhanced authentication' is enabled, the error disappears.
> 
> If the issue persists with 'Enhanced authentication' enabled, please let me know.

Hi @stevkan, I did open the 'Enhanced authentication'

![image](https://user-images.githubusercontent.com/74894646/112399775-d2906080-8d41-11eb-8dc8-57559f7c2194.png)

**Comment by stevkan** [comment]
In addition to 'Enhanced authentication' being enabled, you also need to specify the hosting page of your Web Chat client as a trusted origin (see below). This allows OAuth requests coming from the user via the client to be authenticated without the need of a magic code.

When I tested without the hosting page having been listed as a trusted origin, but with 'Enhanced authentication enabled, it would require a magic code. When I tested with the hosting page listed, no magic code was displayed.

![image](https://user-images.githubusercontent.com/18403946/112894449-ba497a00-9090-11eb-9698-c6c56a1dfa4c.png)

**Comment by CharingJC** [comment]
[index.zip](https://github.com/microsoft/BotFramework-WebChat/files/6247279/index.zip)
 Here is the webchat file i used to connect to my bot

**Comment by CharingJC** [comment]
> In addition to 'Enhanced authentication' being enabled, you also need to specify the hosting page of your Web Chat client as a trusted origin (see below). This allows OAuth requests coming from the user via the client to be authenticated without the need of a magic code.
> 
> When I tested without the hosting page having been listed as a trusted origin, but with 'Enhanced authentication enabled, it would require a magic code. When I tested with the hosting page listed, no magic code was displayed.
> 
> ![image](https://user-images.githubusercontent.com/18403946/112894449-ba497a00-9090-11eb-9698-c6c56a1dfa4c.png)

After adding port that I used for localhost, Chrome is working without require magic code. However Edge have different behavior and still asking for magic code.

Wechat: 4.12.1
Edge version: 
![image](https://user-images.githubusercontent.com/74894646/113372774-f8e67980-939b-11eb-9551-e850e1e193d1.png)

**Comment by dmvtech** [comment]
I'm still getting the magic code for both Edge _and Chrome_.

**Chrome:** Version 89.0.4389.114 (Official Build) (64-bit)
**Edge:** Version 89.0.774.68 (Official build) (64-bit)

@CharingJC Did you do anything specific to get it working in Chrome? or it just worked for you without any intervention?

I will continue to investigate with @stevkan and will update.

**Comment by stevkan** [comment]
Closing due to inactivity. If the issue persists or you have new details to share, please feel free to reopen.

**Comment by aurelvasilegm** [comment]
Hello,
We have the same problem in our environment. We have enabled directline enhanced authentication, we have added our container page URL to trusted domains, but we receive the magic code.
Tested in **Chrome** ver. 101.0.4951.67 and **Edge** ver. 101.0.1210.53.

Direct line channel configuration is:
![image](https://user-images.githubusercontent.com/106539101/171010051-4efd7902-e04c-4b23-b355-2ed2230b04d1.png)

The body of our conversation token is:
{
  "bot": "...",
  "site": "3ylg_kdPhRQ",
  "conv": "7mvRpRKvVKHLBnRkVkfCVE-fr",
  "user": "dl_anonymous",
  "origin": "http://localhost:4200",
  "nbf": 1653917342,
  "exp": 1653920942,
  "iss": "https://directline.botframework.com/",
  "aud": "https://directline.botframework.com/"
}
